### ISSUE TITLE
bug in ket normalization check

### ISSUE THREAD

**Issue by meeg** [sequence-toolbox/SeQUeNCe#155]
I think this line has a math mistake:

https://github.com/sequence-toolbox/SeQUeNCe/blob/9e12549afb3fef0828eb38a48192a906ad7dcc81/src/kernel/quantum_state.py#L88

Unless I'm missing something, what we want is to check that the sum of squared norms is 1. But we are actually checking that the sum of squares is 1. Since the amplitudes are complex, these are not equivalent - e.g. this assert fails for `(1j, 0)` which should be a valid state.

in other words:

```assert abs(sum([a ** 2 for a in amplitudes]) - 1) < 1e-5```
should be 
```assert abs(sum([abs(a) ** 2 for a in amplitudes]) - 1) < 1e-5```

**Comment by Alex-Kolar** [comment]
Hi meeg,

Thanks for pointing this out. I made the fix on a new branch and created pull request #156. There are a few other fixes planned with this pull but it should be completed shortly.
